src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -407,15 +411,17 @@ textToMode ("ignore" : [x])
 textToMode _         = NotAnAnnotation
 
 parseFileInfo :: MarkdownConfig -> FilePath -> LT.Text -> (FileInfo, [ScanError])
-parseFileInfo config fp input
+parseFileInfo config shownFilepath input
   = runWriter
   $ flip runReaderT config
-  $ nodeExtractInfo fp
+  $ nodeExtractInfo shownFilepath
   $ commonmarkToNode [optFootnotes] [extAutolink]
   $ toStrict input
 
 markdownScanner :: MarkdownConfig -> ScanAction
-markdownScanner config path = parseFileInfo config path . decodeUtf8 <$> BSL.readFile path
+markdownScanner config canonicalRoot canonicalFile =
+  parseFileInfo config (getRelativeOrAbsoluteChild canonicalRoot canonicalFile) . decodeUtf8
+    <$> BSL.readFile (unCanonicalPath canonicalFile)
```

Comment:
I think that the way of obtaining `shownName` value will not be different depending on the file format, it seems that it would make more sense just to pass `shownName` right away.